build/scripts/GitHub/GitHubWorkitemLink.class.psm1

```diff
@@ -0,0 +1,70 @@
+
+class GitHubWorkitemLink {
+
+    <#
+        Gets the linked issue IDs from the description.
+        .returns
+            An array of linked issue IDs.
+    #>
+    static [int[]] GetLinkedIssueIDs($Description) {
+        if(-not $Description) {
+            return @()
+        }
+
+        $workitemPattern = "(^|\s)(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved) #(?<ID>\d+)" # e.g. "Fixes #1234"
+        return [GitHubWorkitemLink]::GetLinkedWorkItemIDs($workitemPattern, $Description)
+    }
+
+    <#
+        Gets the linked ADO workitem IDs from the description.
+        .returns
+            An array of linked ADO workitem IDs.
+    #>
+    static [int[]] GetLinkedADOWorkitems($Description) {
+        if(-not $Description) {
+            return @()
+        }
+
+        $workitemPattern = "AB#(?<ID>\d+)" # e.g. "AB#1234" or "Fixes AB#1234"
+        return [GitHubWorkitemLink]::GetLinkedWorkItemIDs($workitemPattern, $Description)
+    }
+
+    <#
+        Links the pull request to the ADO workitem.
+        .returns
+            The updated description.
+    #>
+    static [string] LinkToADOWorkItem($Description, $WorkItem) {
+        if ($Description -match "AB#$($WorkItem)") {
+            Write-Host "Pull request already linked to ADO workitem AB#$($WorkItem)"

```

Comment:
Why is "Pull Request" inferred? 
Maybe `LinkToADOWorkItem` can return _true_/_false_ depending if it updated `$Description`, which can be passed as _ref_?